x/did/client/rest/tx.go

```diff
@@ -1,90 +1,58 @@
 package rest
 
 import (
-	"fmt"
-	"github.com/cosmos/cosmos-sdk/types/rest"
-	"net/http"
-	"time"
-
 	"github.com/cosmos/cosmos-sdk/client/context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/rest"
 	"github.com/gorilla/mux"
+	"github.com/ixofoundation/ixo-blockchain/x/did/exported"
+	"net/http"
 
-	"github.com/ixofoundation/ixo-blockchain/x/did/internal/keeper"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
 	"github.com/ixofoundation/ixo-blockchain/x/did/internal/types"
-	"github.com/ixofoundation/ixo-blockchain/x/ixo"
 )
 
 func registerTxRoutes(cliCtx context.CLIContext, r *mux.Router) {
-	r.HandleFunc("/did", createDidRequestHandler(cliCtx)).Methods("POST")
-	r.HandleFunc("/credential", addCredentialRequestHandler(cliCtx)).Methods("POST")
+	r.HandleFunc("/did/add_did", addDidRequestHandler(cliCtx)).Methods("POST")
+	r.HandleFunc("/did/add_credential", addCredentialRequestHandler(cliCtx)).Methods("POST")
 }
 
-func createDidRequestHandler(cliCtx context.CLIContext) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-
-		w.Header().Set("Content-Type", "application/json")
-		didDocParam := r.URL.Query().Get("didDoc")
-		mode := r.URL.Query().Get("mode")
-		cliCtx = cliCtx.WithBroadcastMode(mode)
-
-		ixoDid, err := types.UnmarshalIxoDid(didDocParam)
-		if err != nil {
-			w.WriteHeader(http.StatusBadRequest)
-			_, _ = w.Write([]byte(err.Error()))
-			return
-		}
-
-		msg := types.NewMsgAddDid(ixoDid.Did, ixoDid.VerifyKey)
+type AddDidReq struct {
+	BaseReq rest.BaseReq `json:"base_req" yaml:"base_req"`
+	Did     exported.Did `json:"did" yaml:"did"`
+	PubKey  string       `json:"pubKey" yaml:"pubKey"`
+}
 
-		output, err := ixo.CompleteAndBroadcastTxRest(cliCtx, msg, ixoDid)
-		if err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			_, _ = w.Write([]byte(err.Error()))
+func addDidRequestHandler(cliCtx context.CLIContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var req AddDidReq
+		req.BaseReq = req.BaseReq.Sanitize()
```

Comment:
Could you please compare the approach with Cosmos SDK once again please? I think we are missing `rest.ReadRESTReq` in all of the functions. `BaseReq` will not be anything until we read the request.

Reply:
This applies to all of the handlers in this PR.

Reply:
yeah got it. updated it to all modules